vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -560,7 +561,7 @@ def make_batch(
         if resource_type == vim.ClusterComputeResource:
             # Cluster metrics are unpredictable and a single call can max out the limit. Always collect them one by one.
             max_batch_size = 1  # type: float
-        elif resource_type in REALTIME_RESOURCES or self._config.max_historical_metrics < 0:
+        elif is_historical or self._config.max_historical_metrics < 0:
```

Comment:
```suggestion
        elif resource_type in RESOURCES_WITH_HISTORICAL_METRICS or self._config.max_historical_metrics < 0:
```
Not sure about the values of `resource_type` but we I guess we could do something like this.

Reply:
`is_historical` is referring to whether the batch metric request is going to be for historical or realtime metrics. Since now resources can have both realtime and historical metrics, if we use this logic, we could get some batches that contain both historical and realtime metrics, which we do not want (could negatively impact performance). I updated the variable to be called `is_historical_batch`, let me know if that helps